nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -254,20 +254,20 @@ let
             else defaultListen;
 
         listenString = { addr, port, ssl, extraParameters ? [], ... }:
-          "listen ${addr}:${toString port} "
-          + optionalString ssl "ssl "
+          (if ssl && vhost.http3 then "
+          # UDP listener for **QUIC+HTTP/3
+          listen ${addr}:${toString port} http3 "
+          + optionalString vhost.default "default_server "
+          + optionalString vhost.http3Reuseport "reuseport"
```

Comment:
`reuseport` isn't http3 specific, why don't we enable ability to set `reuseport` for all http protocols?

Reply:
For other http protocols , reuseport can be set like this:
```
services.nginx.virtualHosts."example.local".listen.*.extraParameters = [ "reuseport" ];
```
For the http3 protocol, this parameter is required and must be specified once.

Reply:
> services.nginx.virtualHosts."example.local".listen.*.extraParameters = [ "reuseport" ];

I just think it's more reasonable to write 

```
services.nginx.virtualHosts."example.local" = {
  addSSL = true;
  reuseport = true;
};
```

Instead of
```
services.nginx.virtualHosts."example.local" = {
  addSSL = true;
  listen = [
    {
        addr = "0.0.0.0";
        port = 443;
        ssl = true;
        extraParameters = [ "reuseport" ];
    }
    {
        addr = "[::0];
        port = 443;
        ssl = true;
        extraParameters = [ "reuseport" ];
    }
    {
        addr = "0.0.0.0";
        port = 80;
        extraParameters = [ "reuseport" ];
    }
    {
        addr = "[::0];
        port = 80;
        extraParameters = [ "reuseport" ];
    }
  ];
};
```

Reply:
Updated PR.